src/components/questionnaire/new-questionnaire-builder.tsx

```diff
@@ -175,26 +172,49 @@ const NewQuestionnaireBuilder = () => {
    * Function to save the new questionnaire
    */
   const saveQuestionnaire = async () => {
+    // Enhanced validation before saving
+    if (!questionnaire.title || !questionnaire.description) {
+      setError(strings.error.questionnaireSaveFailed + ", Title and description are required");
+      return;
+    }
+    if (!questionnaire.questions || questionnaire.questions.length === 0) {
+      setError(strings.error.questionnaireSaveFailed + ", At least one question is required");
+      return;
+    }
+    const hasAnyCorrectAnswers = questionnaire.questions.some(question => 
+      question.answerOptions && question.answerOptions.some(option => option.isCorrect)
+    );
+
+    if (!hasAnyCorrectAnswers) {
+      setError(strings.error.questionnaireSaveFailed + ", At least one answer must be marked as correct");
+      return;
+    }
+    // Check if passScore is set
+    if (questionnaire.passScore === undefined || questionnaire.passScore === null) {
+      setError(strings.error.questionnaireSaveFailed + ", Please set a required score using the slider");
+      return;
+    }
```

Comment:
This is good validation but is it needed here at the point of saving? It would be better if the save button was disabled based on these conditions (and a tooltip should be added to the disabled save button if they are not met- showing just the single message working down from the top e.g. if all conditions are not met it only needs to say title and description are missing).

Also as these errors are being shown to the user they should be localized using the strings package we already have set up, not hardcoded strings in english.

Also these should be placed in a single validateQuestionnaire function for clarity (and it would probably make sense for this to be exported from the questionnaireBuilderUtils file you created).